tests/integration/pox/pox-3/direct-and-pooled.spec.ts

```diff
@@ -0,0 +1,283 @@
+    );
+    expect(response.error).toBeUndefined();
+
+    // Alice delegates 100m to Bob, but it fails.
+    response = await broadcastDelegateSTX(
+      {
+        poxVersion: 3,
+        network,
+        account: Accounts.WALLET_1,
+        fee,
+        nonce: aliceNonce++,
+      },
+      { amount: 100_000_000_000_000, poolAddress: Accounts.WALLET_2 }
+    );
+    expect(response.error).toBeUndefined();
+    let [block, tx] = await waitForStacksTransaction(
+      orchestrator,
+      response.txid
+    );
+    expect(tx.success).toBeFalsy();
+    expect(tx.result).toBe("(err 3)");
+
+    let poxInfo = await getPoxInfo(network);
+
+    // Assert that the next cycle has 75m STX locked
+    expect(poxInfo.current_cycle.id).toBe(1);
+    expect(poxInfo.current_cycle.min_threshold_ustx).toBe(29_290_000_000_000);
+    expect(poxInfo.next_cycle.stacked_ustx).toBe(75_000_000_000_000);
+
+    // Check Alice's table entry
+    let poxAddrInfo0 = await readRewardCyclePoxAddressForAddress(
+      network,
+      3,
+      2,
+      Accounts.WALLET_1.stxAddress
+    );
+    expect(poxAddrInfo0?.["total-ustx"]).toEqual(uintCV(75_000_000_000_000));
+  });
+
+  it("stacking more should increase the slot price (cycle #1)", async () => {
+    const network = new StacksTestnet({ url: orchestrator.getStacksNodeUrl() });
+    const cycles = 1;
+
+    // Faucet stacks 900m (1/4 of liquid suply)
+    let response = await broadcastStackSTX(
+      {
+        poxVersion: 3,
+        network,
+        account: Accounts.FAUCET,
+        fee,
+        nonce: faucetNonce++,
+      },
+      { amount: 900_000_000_000_000, blockHeight, cycles }
+    );
+    expect(response.error).toBeUndefined();
+
+    await asyncExpectStacksTransactionSuccess(orchestrator, response.txid);
+
+    let poxInfo = await getPoxInfo(network);
+
+    // Assert that the next cycle has 975m STX locked
+    // and the slot price increased above Alice's stacking
+    expect(poxInfo.current_cycle.id).toBe(1);
+    expect(poxInfo.current_cycle.min_threshold_ustx).toBe(29_290_000_000_000);
+    expect(poxInfo.next_cycle.stacked_ustx).toBe(975_000_000_000_000);
+  });
+
+  it("auto-unlock unlocks directly stacked amount and total_stacked is updated (cycle #2)", async () => {
+    const network = new StacksTestnet({ url: orchestrator.getStacksNodeUrl() });
+
+    // Wait for Stacks genesis block
+    await orchestrator.waitForNextStacksBlock();
+
+    // move on to the next cycle after unlock (N+5)
+    await waitForNextRewardPhase(network, orchestrator, 5);
+
+    let poxInfo = await getPoxInfo(network);
+    // Assert that the current cycle has 900m STX locked from direct stacking
+    expect(poxInfo.current_cycle.id).toBe(2);
+    expect(poxInfo.current_cycle.stacked_ustx).toBe(900_000_000_000_000);
+    // FIXME: Is this right? Alice has stacked 75M and she gets unlocked
+    expect(poxInfo.current_cycle.min_threshold_ustx).toBe(75_000_000_000_000);
```

Comment:
This needs to be double-checked. The min threshold is reported as 75M. Alice gets auto-unlocked with 75M.